eth/vm/forks/london/transactions.py

```diff
@@ -0,0 +1,324 @@
+)
+from rlp.sedes import (
+    CountableList,
+    big_endian_int,
+    binary,
+)
+
+from .constants import BASE_GAS_FEE_TRANSACTION_TYPE
+
+
+class LondonLegacyTransaction(BerlinLegacyTransaction):
+    pass
+
+
+class LondonUnsignedLegacyTransaction(BerlinUnsignedLegacyTransaction):
+    def as_signed_transaction(self,
+                              private_key: PrivateKey,
+                              chain_id: int = None) -> LondonLegacyTransaction:
+        v, r, s = create_transaction_signature(self, private_key, chain_id=chain_id)
+        return LondonLegacyTransaction(
+            nonce=self.nonce,
+            gas_price=self.gas_price,
+            gas=self.gas,
+            to=self.to,
+            value=self.value,
+            data=self.data,
+            v=v,
+            r=r,
+            s=s,
+        )
+
+
+
+class UnsignedBaseGasFeeTransaction(rlp.Serializable):
+    _type_id = BASE_GAS_FEE_TRANSACTION_TYPE
+    fields = [
+        ('chain_id', big_endian_int),
+        ('nonce', big_endian_int),
+        ('max_priority_fee_per_gas', big_endian_int),
+        ('max_fee_per_gas', big_endian_int),
+        ('gas', big_endian_int),
+        ('to', address),
+        ('value', big_endian_int),
+        ('data', binary),
+        ('access_list', CountableList(AccountAccesses)),
+    ]
+
+    @cached_property
+    def _type_byte(self) -> bytes:
+        return to_bytes(self._type_id)
+
+    def get_message_for_signing(self) -> bytes:
+        payload = rlp.encode(self)
+        return self._type_byte + payload
+
+    def as_signed_transaction(self, private_key: PrivateKey) -> 'TypedTransaction':
+        message = self.get_message_for_signing()
+        signature = private_key.sign_msg(message)
+        y_parity, r, s = signature.vrs
+
+        signed_transaction = BaseGasFeeTransaction(
+            self.chain_id,
+            self.nonce,
+            self.max_priority_fee_per_gas,
+            self.max_fee_per_gas,
+            self.gas,
+            self.to,
+            self.value,
+            self.data,
+            self.access_list,
+            y_parity,
+            r,
+            s
+        )
+        return LondonTypedTransaction(self._type_id, signed_transaction)
+
+
+class BaseGasFeeTransaction(rlp.Serializable, SignedTransactionMethods, SignedTransactionAPI):
+    _type_id = BASE_GAS_FEE_TRANSACTION_TYPE
+    fields = [
+        ('chain_id', big_endian_int),
+        ('nonce', big_endian_int),
+        ('max_priority_fee_per_gas', big_endian_int),
+        ('max_fee_per_gas', big_endian_int),
+        ('gas', big_endian_int),
+        ('to', address),
+        ('value', big_endian_int),
+        ('data', binary),
+        ('access_list', CountableList(AccountAccesses)),
+        ('y_parity', big_endian_int),
+        ('r', big_endian_int),
+        ('s', big_endian_int),
+    ]
+
+    @property
+    def gas_price(self) -> None:
+        # maybe add a warning, or raise an exception instead?
+        return None
```

Comment:
How should we handle this? `gas_price` is no longer a transaction attribute. It could still be interpreted as "effective gas price", but that requires knowing the base gas fee of the block so it would not a be "transaction-only" attribute anymore